src/App.jsx

```diff
@@ -1,23 +1,27 @@
-import React from 'react';
-import { Routes, Route } from 'react-router-dom';
+import {Routes, Route, Navigate} from 'react-router-dom';
 import NavBar from './components/NavBar';
 import Profile from './pages/Profile';
 import Home from './pages/Home';
 import SignIn from './pages/SignIn';
 import SignUp from './pages/SignUp';
 import './App.css';
+import {AuthContext} from './context/AuthContext';
+import {useContext} from "react";
 
 function App() {
+    const {state} = useContext(AuthContext);
+
   return (
     <>
       <NavBar />
       <div className="content">
         <Routes>
           <Route path="/" element={<Home />} />
-          <Route path="/profile" element={<Profile />} />
+          <Route path="/profile" element={state === true ? <Profile/> : <Navigate to="/SignIn"/>}/>
```

Comment:
Je kan de expliciete check hier ook versimpelen tot een impliciete check

```suggestion
          <Route path="/profile" element={state ? <Profile/> : <Navigate to="/SignIn"/>}/>
```